src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -335,6 +425,37 @@ void ControllerScriptInterfaceLegacy::triggerScriptConnection(
     connection.executeCallback(coScript->get());
 }
 
+bool ControllerScriptInterfaceLegacy::removeRuntimeDataConnection(
+        const ScriptConnection& connection) {
+    VERIFY_OR_DEBUG_ASSERT(m_pScriptEngineLegacy->jsEngine()) {
+        return false;
+    }
+    return m_runtimeDataConnections.removeAll(connection) > 0;
+}
+
+void ControllerScriptInterfaceLegacy::triggerRuntimeDataConnection(
+        const ScriptConnection& connection) {
+    VERIFY_OR_DEBUG_ASSERT(m_pScriptEngineLegacy->jsEngine() ||
+            !m_runtimeDataConnections.contains(connection)) {
+        return;
+    }
+    auto pJsEngine = m_pScriptEngineLegacy->jsEngine();
+
+    QJSValue func = connection.callback; // copy function because QJSValue::call is not const
+    auto args = QJSValueList{
+            pJsEngine->toScriptValue(m_pScriptEngineLegacy->getSharedData()->get()),
+    };
+    QJSValue result = func.call(args);
+    if (result.isError()) {
+        if (m_pScriptEngineLegacy != nullptr) {
+            m_pScriptEngineLegacy->showScriptExceptionDialog(result);
+        }
+        qWarning() << "ControllerEngine: Invocation of runtime data connection "
+                   << connection.id.toString()
+                   << "connected to runtime data failed:" << result.toString();
+    }
```

Comment:
how about you throw a JS exception instead?

Reply:
Updated this one and the one above